vllm/engine/async_llm_engine.py

```diff
@@ -700,3 +696,22 @@ async def check_health(self) -> None:
         else:
             await self.engine.check_health_async()
         logger.debug(f"Health check took {time.perf_counter()-t}s")
+
+    def _shorten_for_logging(self, prompt,
+                             prompt_token_ids) -> Tuple[str, str]:
+        """Truncates the input prompt text and token ids for logging.
+        The string representations of both are truncated to a maximum length
+        of self.max_log_len, and then ellipses (...) are added to indicate
+        that the values were truncated.
+        """
+        if self.max_log_len is not None:
+            if prompt is not None and len(prompt) > self.max_log_len:
+                prompt = f"{prompt:.{self.max_log_len}}..."
+            if prompt_token_ids is not None:
+                # This regex removes the last value in the list after the
+                # string representation is truncated, to avoid logging an
+                # incorrect token id.
+                prompt_token_ids = re.sub(
+                    self.list_truncation_regex, ', ...]',
+                    f"{prompt_token_ids!s:.{self.max_log_len}}")
```

Comment:
I'd be wary about using a regex here because of the performance overhead. Maybe it would be better to either divide the length by 6 or have a separate parameter for configuring the max number of tokens to log?

Reply:
Based on https://github.com/vllm-project/vllm/issues/1240#issuecomment-2031579826, maybe I'll go with adding new flags to disable logging the tokens and/or request parameters

Reply:
@joerunde how about leaving the current logic i.e. `max_log_len` controls both text and tokens, but just add ellipses (in token case could do `shortened_token_ids[-1] = "..."`), and also special handling for `max_log_len==0` (disable logging tokens and text)?

Reply:
🌶️ , pivoted away from this for a bit but got those changes for ya